Rubberduck.Parsing/VBA/ModuleState.cs

```diff
@@ -20,6 +21,8 @@ public class ModuleState
         public List<IAnnotation> Annotations { get; private set; }
         public SyntaxErrorException ModuleException { get; private set; }
         public IDictionary<Tuple<string, DeclarationType>, Attributes> ModuleAttributes { get; private set; }
+        public ConcurrentDictionary<QualifiedModuleName, byte> HasReferenceToModule { get; private set; }
+        public HashSet<QualifiedModuleName> IsReferencedByModule { get; private set; }
```

Comment:
Aren't the collection types reversed? If the collections populated as the resolver did its thing, wouldn't `HasReferencesTo` be populated by the thread walking that parse tree, and `IsReferencedBy` would be write-accessed by each other modules' threads, no? Could be a `ConcurrentDictionary<QualifiedModuleName, byte>`  where the byte is just a dummy - shame .net has no `ConcurrentHashSet<T>`!